docs: add metadata to all remaining pages (#1246)
canonical/rockcraft: docs/reference/remote-builds.rst

.. meta::

>     :description: Reference documentation for Rockcraft remote builds via Launchpad: how remote builds work, supported architectures, and the remote-build command.

Reviewer: ```suggestion
    :description: Reference documentation for remote rock builds on Launchpad. How remote builds work, the supported architectures, and details about the remote build command.
```
